### DISCUSSION TITLE
ASCII Code Figuren programmiert

### DISCUSSION THREAD

**Discussion by Wildeswiesel** [Wildeswiesel/Space-Waffel#3] [General]
Am 18.12 haben wir in Python die For-Schleife kennengelernt. Nach kurzer Einführung durften wir auch schon selber coden, wobei die meisten VS Code verwenden. Am anfang hatte wir die aufgabe ein ASCII Art zu erstellen z.b. 2 Dreiecke welche in unterschiedlichen zeichen sind. Danach hatten wir noch etwas zeit wo ich einen Tannen baum erstellt habe. 
z.b. Den Tannenbaum Code:
[https://github.com/Wildeswiesel/Space-Waffel/blob/Test/ASCII_Tannenbaum.py](url)
```
höhe=8
stammhöhe=2
for i in range(höhe):
    spaces=" " * (höhe-i-1)
    characters="x" * (2*i+1)
    print(spaces + characters)

for j in range(stammhöhe):
    ospaces=" "*(höhe-2)
    print(ospaces + "ooo")
```

Dieser sah dann wie folgt aus:

![image](https://github.com/Wildeswiesel/Space-Waffel/assets/152853691/afb9012e-88c3-465b-a3ce-28c6cfdeea80)

Des Weitern haben wir mit Turtle verschiedene Bilder gezeichnet

Dort habe ich eine Tasse erstellt 
[https://github.com/Wildeswiesel/Space-Waffel/blob/Test/tasse_turtle.py](url)
```
import turtle

wn = turtle.Screen()
wn.bgcolor("white")
wn.title("Turtle")
t = turtle.Turtle()
t.speed(10)
t.hideturtle

def tasse():
    t.right(90)
    t.forward(40)
    for i in range(18):
        t.right(5)
        t.forward(2)

    t.forward(30)

    for j in range(18):
        t.right(5)
        t.forward(2)

    t.forward(40)


def henkel():
    t.right(180)
    t.forward(10)
    t.right(90)
    for i in range(62):
        t.forward(1)
        t.left(3)

tasse()
henkel()
turtle.done()
```
Dieses sah dann wie folgt aus 

![image](https://github.com/Wildeswiesel/Space-Waffel/assets/152853691/05f9b757-5a22-469a-80b2-cfb09a597ed4)

**Comment by SvenSrc** [comment]
Your example is fitting to our time of the year with the Christmas Tree.

**Comment by HamidrezaRahimian** [comment]
wow , good job 
thanks for sharing the text

**Comment by stechmo** [comment]
Schön mit Funktionen realisiert, sauberer Code zu lesen macht immer Spaß!
